packages/core/test/financial-templates/common/financial-product-libraries/long-short-pair-libraries/CappedYieldDollarLongShortPairFinancialProductLibrary.js

```diff
@@ -0,0 +1,104 @@
+        .send({ from: accounts[0] });
+
+      const setLowPriceRange = await cappedYieldDollarLSPFPL.methods.lowPriceRanges(lspMock.options.address).call();
+      assert.equal(setLowPriceRange.toString(), lowPriceRange);
+    });
+    it("Can not re-use existing LSP contract address", async () => {
+      await cappedYieldDollarLSPFPL.methods
+        .setLongShortPairParameters(lspMock.options.address, lowPriceRange)
+        .send({ from: accounts[0] });
+
+      // Second attempt should revert.
+      assert(
+        await didContractThrow(
+          cappedYieldDollarLSPFPL.methods
+            .setLongShortPairParameters(lspMock.options.address, lowPriceRange)
+            .send({ from: accounts[0] })
+        )
+      );
+    });
+    it("Can not set invalid LSP contract address", async () => {
+      // LSP Address must implement the `expirationTimestamp method.
+      assert(
+        await didContractThrow(
+          cappedYieldDollarLSPFPL.methods
+            .setLongShortPairParameters(ZERO_ADDRESS, lowPriceRange)
+            .send({ from: accounts[0] })
+        )
+      );
+    });
+  });
+  describe("Compute expiry tokens for collateral", () => {
+    beforeEach(async () => {
+      await cappedYieldDollarLSPFPL.methods
+        .setLongShortPairParameters(lspMock.options.address, lowPriceRange)
+        .send({ from: accounts[0] });
+    });
+    it("Lower than low price range should return 1 (long side is short put option)", async () => {
+      // If the price is lower than the low price range then the max payout per each long token is hit at the full
+      // collateralPerPair. i.e each short token is worth 0*collateralPerPair and each long token is worth 1*collateralPerPair.
+      const expiryTokensForCollateral = await cappedYieldDollarLSPFPL.methods
+        .percentageLongCollateralAtExpiry(toWei("9"))
+        .call({ from: lspMock.options.address });
+      assert.equal(expiryTokensForCollateral.toString(), toWei("1"));
+    });
+    it("Above the low price range should return long worth bond notional (long side is long yield dollar)", async () => {
+      // If the price is above the low price range then the payout is simply that of a yield dollar. i.e every
+      // long token is worth the bond notional of 100. At a price of 20 we are between the bounds. Each long should be worth
+      // 100 so there should be 100/20=5 UMA per long token. As each collateralPerPair is worth 10, expiryPercentLong should
+      // be 10/5=0.5, thereby allocating half to the long and half to the short.
+      const expiryTokensForCollateral1 = await cappedYieldDollarLSPFPL.methods
+        .percentageLongCollateralAtExpiry(toWei("20"))
+        .call({ from: lspMock.options.address });
+      assert.equal(expiryTokensForCollateral1.toString(), toWei("0.5"));
+
+      // Equally, at a price of 40 each long should still be worth 100 so there should be 100/40=2.5 UMA per long. As
```

Comment:
this test does not do what your comments say it does. you say to enter 40 as the price and check the value is 0.25 from the lib such that value equals 100 but the test below uses 20 and checks for 0.5

Reply:
Good catch, will fix that comment.

Reply:
I found a couple of additional errors in these comments and will fix them, too.